sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 9/11/2022.
+ */
+
+package com.adyen.checkout.sessions.provider
+
+import com.adyen.checkout.components.base.Configuration
+import com.adyen.checkout.sessions.CheckoutSession
+import com.adyen.checkout.sessions.model.SessionModel
+import kotlinx.coroutines.CoroutineScope
+
+// TODO docs
+object CheckoutSessionProvider {
+    // TODO docs
+    suspend fun createSession(
+        sessionModel: SessionModel,
+        configuration: Configuration,
+    ): Result<CheckoutSession> {
+        return CheckoutSessionInitializer(sessionModel, configuration).setupSession()
+    }
+
+    // TODO docs
+    // callback alternative for the suspend function
+    fun createSession(
```

Comment:
If we are giving a callback alternative it should also be without `CoroutineScope` IMO. Now it's basically the same thing, but we are calling 'launch'. WDYT?

Reply:
How do you suggest we do it without using any `CoroutineScope`?

Reply:
With `ThreadManager` 😂 No, let's not do that. To be honest I'm not sure what the best way of doing this is

Reply:
😂 ☠️